infra/pr_helper.py

```diff
@@ -0,0 +1,246 @@
+    if email_response[0] is not None:
+      if is_known_contributor(content_dict, email_response[1]):
+        # Checks if the email is verified.
+        if email_response[0]:
+          message += (
+              f'@{pr_author} is either the primary contact or '
+              f'is in the CCs list of [{project_path}]({project_url}).<br/>')
+          continue
+        message += (f'@{pr_author} is either the primary contact or '
+                    f'is in the CCs list of [{project_path}]({project_url}), '
+                    f'but their email {email_response[1]} '
+                    'is not verified.<br/>')
+
+    # Checks the previous commits.
+    has_commit = github.has_author_modified_project(project_path)
+    if not has_commit:
+      message += (
+          f'@{pr_author} is a new contributor to '
+          f'[{project_path}]({project_url}). The PR must be approved by known '
+          'contributors before it can be merged.<br/>')
+      is_ready_for_merge = False
+      continue
+    commit_sha = has_commit[1]
+
+    # If the previous commit is not associated with a pull request.
+    pr_message = (f'@{pr_author} has previously contributed to '
+                  f'[{project_path}]({project_url}). The previous commit was '
+                  f'{GITHUB_URL}/{OWNER}/{REPO}/commit/{commit_sha}<br/>')
+
+    pr_url = github.get_pull_request_url(commit_sha)
+    if pr_url is not None:
+      pr_message = (f'@{pr_author} has previously contributed to '
+                    f'[{project_path}]({project_url}). '
+                    f'The previous PR was {pr_url}<br/>')
+    message += pr_message
+
+  save_env(message, is_ready_for_merge, False)
+
+
+class GithubHandler:
+  """Github requests handler."""
+
+  def __init__(self):
+    self._pr_author = os.environ['PRAUTHOR']
+    self._token = os.environ['GITHUBTOKEN']
+    self._pr_number = os.environ['PRNUMBER']
+    self._headers = {
+        'Authorization': f'Bearer {self._token}',
+        'X-GitHub-Api-Version': '2022-11-28'
+    }
+    os.environ['GITHUB_AUTH_TOKEN'] = self._token
+
+  def get_pr_author(self):
+    """Gets the pr author user name."""
+    return self._pr_author
+
+  def get_projects_path(self):
+    """Gets the current project path."""
+    response = requests.get(f'{BASE_URL}/pulls/{self._pr_number}/files',
+                            headers=self._headers)
+
+    projects_path = set()
+    for file in response.json():
+      file_path = file['filename']
+      dir_path = os.path.dirname(file_path)
+      if dir_path is not None and dir_path.split(os.sep)[0] == 'projects':
+        projects_path.add(dir_path)
+    return list(projects_path)
+
+  def get_author_email(self):
+    """Retrieves the author's email address for a pull request,
+    including non-public emails."""
+    user_response = requests.get(f'{API_URL}/users/{self._pr_author}')
+    if user_response.ok:
+      email = user_response.json()['email']
+      if email:
+        return True, email
```

Comment:
Is this `if` block implying that `email` might not exist in `user_response.json()`?
If so, use `get('email')` to avoid `KeyError`.

Reply:
The `email` will exist as response, but it may be `None`. The check is to make sure that it's not `None`

Reply:
Ah thanks! Then the convention is to check `None` values explicitly: e.g., `if email is None`.

Reply:
But I want to make sure that the string is not `None` or `empty`.